elements/rh-footer/rh-footer.ts

```diff
@@ -1,18 +1,210 @@
+ * @slot    header - Overrides `header-*`, `logo`, `social-links`
+ * @csspart header - footer header, typically containing main logo and social links
+ * @slot    header-primary - primary footer header content, e.g. main logo. Overrides `logo`
+ * @csspart header-primary - primary footer header content, e.g. main logo
+ * @slot    header-secondary - secondary footer header content, e.g. social links. Overrides `social-links`
+ * @csspart header-secondary - secondary footer header content, e.g. social links
+ * @slot    logo - main page or product logo. Defaults to Red Hat corporate logo
+ * @csspart logo - main page or product logo container
+ * @slot    social-links - social media links (icons). Contains a default set of links
+ * @csspart social-links - social links container `<rh-footer-links>`
+ * @slot    main - main footer content. Overrides `main-*`
+ * @csspart main - main content container.
+ * @slot    main-primary - main footer region. typically a columnar grid
+ * @csspart main-primary - container for main footer links
+ * @slot    links - main footer links
+ * @csspart links - container for main footer links
+ * @csspart links-accordion-header - mobile links accordion header element
+ * @csspart links-accordion-panel - mobile links panel container element
+ * @slot    main-secondary - typically contains prose or promotional content
+ * @csspart main-secondary - container fro prose or promotional content
+ * @slot    global - must contain `<rh-footer-universal>` {@deprecated - use `universal`}
+ * @slot    universal - must contain `<rh-footer-universal>`
+ *
+ * @cssprop --rh-footer-icon-color - {@default #8a8d90}
+ * @cssprop --rh-footer-icon-color-hover - {@default #b8bbbe}
+ * @cssprop --rh-footer-border-color - {@default #6a6e73}
+ * @cssprop --rh-footer-accent-color - {@default #e00}
+ * @cssprop --rh-footer-section-side-gap - {@default 32px}
+ * @cssprop --rh-footer-links-gap - {@default 8px}
+ * @cssprop --rh-footer-link-header-font-size - {@default 0.875em}
+ * @cssprop --rh-footer-nojs-min-height - {@default 750px}
+ */
+@customElement('rh-footer')
+export class RhFooter extends LitElement {
+  static readonly version = '{{version}}';
+
+  static readonly styles = [style, responsiveStyles];
+
+  /**
+   * Isomorphic import.meta.url function
+   * Requires a node.js dom shim that sets window.location
+   */
+  static getImportURL(relativeLocation: string | URL): string | URL {
+    const url = new URL(relativeLocation, import.meta.url);
+    if (url.protocol === 'file:') {
+      return new URL(relativeLocation, window.location.href);
+    }
+    return url;
+  }
+
+  #logger = new Logger(this);
+
+  @colorContextProvider()
+  @property({ reflect: true, attribute: 'color-palette' }) colorPalette: ColorPalette = 'darker';
+
+  override connectedCallback() {
+    super.connectedCallback();
+    // wire up accessibility aria-labels with unordered lists
+    this.updateAccessibility();
+  }
+
+  #compact = false;
+
```

Comment:
```suggestion
  #logger = new Logger(this);

  #compact = false;
  
  @colorContextProvider()
  @property({ reflect: true, attribute: 'color-palette' }) colorPalette: ColorPalette = 'darker';

  override connectedCallback() {
    super.connectedCallback();
    // wire up accessibility aria-labels with unordered lists
    this.updateAccessibility();
  }

```

Obsessive whitespace refactoring :shrug: